### ISSUE TITLE
Encountering Error Code 500 on Database Query

### ISSUE THREAD

**Issue by max-legrand** [dstotijn/go-notion#2]
When the following line executes, it returns back stating an error with code 500 occurs.

`	response, err := client.QueryDatabase(context.Background(), id, query)
`

I'm not particularly sure where the issue is arising, so any help resolving is appreciated!

**Comment by dstotijn** [comment]
Hi! Could you share what your `query` value looks like?

**Comment by max-legrand** [comment]
Yep!
<img width="463" alt="image" src="https://user-images.githubusercontent.com/47373807/118546234-a97ac380-b725-11eb-9ea2-6e039fb1f482.png">

I also tried with a blank query and that also kicked back with the same error. For reference, when I just use standard POST requests it seems to work (e.g. via Postman generated Go code)

**Comment by dstotijn** [comment]
Thanks for the detailed feedback! There was a bug where the various search filter fields were not being omitted from JSON, because they were struct values instead of struct pointers. Fixed in [v0.1.1](https://github.com/dstotijn/go-notion/tree/v0.1.1). **Note:** The fix is a breaking change; you'll have to update your code, e.g. to this:

```
query := notion.DatabaseQuery{
	Filter: notion.DatabaseQueryFilter{
		And: []notion.DatabaseQueryFilter{
			{
				Property: "Done",
				Checkbox: &notion.CheckboxDatabaseQueryFilter{
					Equals: notion.BoolPtr(true),
				},
			},
			{
				Property: "Repeat",
				Select: &notion.SelectDatabaseQueryFilter{
					IsNotEmpty: true,
				},
			},
			{
				Property: "Date",
				Date: &notion.DateDatabaseQueryFilter{
					IsNotEmpty: true,
				},
			},
		},
	},
}
```

**Comment by max-legrand** [comment]
Awesome thank you so much for the super fast assistance!
